lib/ansible/modules/system/known_hosts.py

```diff
@@ -306,13 +306,12 @@ def compute_diff(path, found_line, replace_or_add, state, key):
         'after': '',
     }
     try:
-        inf = open(path, "r")
+        with open(path, "r") as inf:
+            diff['before'] = inf.read()
     except IOError as e:
```

Comment:
You are slightly changing semantics here, but I think this is ok.

Before, `IOError`s were only caught when they were raised by `open()`. Now they are also caught when raised by `read()` or `close()`.

Reply:
Yes. Agreed. It's a result of using context managers to handle files.